core/zoe/contracts/helpers/userFlow.js

```diff
@@ -2,11 +2,12 @@ import harden from '@agoric/harden';
 
 export const rejectOffer = (
   zoe,
-  offerHandle,
+  assays,
+  inviteHandle,
   message = `The offer was invalid. Please check your refund.`,
 ) => {
-  zoe.complete(harden([offerHandle]));
-  return Promise.reject(new Error(`${message}`));
+  zoe.complete(harden([inviteHandle]), assays);
```

Comment:
Why does `complete` need an `assays` array? Doesn't it already know this according to the per-contract facet it is?

Reply:
I've been trying to get away from having a canonical set of assays per instance, so I changed the parts of Zoe that assume that. It simplified state.js and the zoe code, although it does make it more complex for smart contracts that work fine with a canonical set of assays per instance. Is there a good way to get out of this tradeoff?